sdk/java-sdk-spring/src/main/java/kalix/javasdk/client/ViewCallBuilder.java

```diff
@@ -40,28 +40,40 @@
 import akka.japi.function.Function9;
 import com.google.protobuf.any.Any;
 import kalix.javasdk.DeferredCall;
+import kalix.javasdk.MetadataContext;
+import kalix.javasdk.action.ActionContext;
 import kalix.javasdk.impl.client.MethodRefResolver;
 import kalix.javasdk.impl.client.ViewCallValidator;
 import kalix.spring.KalixClient;
 
 import java.lang.reflect.Method;
 import java.util.List;
+import java.util.Optional;
 
 public class ViewCallBuilder {
 
   private final KalixClient kalixClient;
 
+  private Optional<MetadataContext> tracingContext = Optional.empty();
+
+
   public ViewCallBuilder(KalixClient kalixClient) {
     this.kalixClient = kalixClient;
   }
 
+  public ViewCallBuilder withTracing(ActionContext context){
```

Comment:
It was a proposal but happy to use `MetadataContext`.  The idea was to signal to the user that this feature only works with Actions at the moment. Once we include workflows I would change it to `MetadataContext`. But I understand from your comment that you are inclined to use it already instead of `ActionContext`.

Reply:
The situation here is slightly different than in the proto-sdk, where we can introduce a change in the generated code more seamlessly for users. This will become a public API, so changing the signature later will be more problematic. I would verify at this point if it is problematic to support workflows or not. If we know that we don't want to support tracing in workflows then `ActionContext` is the correct input. Otherwise, we should consider changing the signature.

Reply:
Right. From what I can see user's code will not perceive any change if we start wit `ActionContext` and then move to `MetadataContext`. They won't have to change anything, as the former extends the latter. Is there anything else you have in mind? I'm curious. 
Another option is to use `MetadataContext` now and add some comment in the java docs stating that it doesn't work in Workflows at the moment. Not ideal I'd say. 
We are definitively interested in add tracing to workflows. What do you think, as a user would you be interested in the feature?

Reply:
Ok, so let's talk about the pros and cons of supporting tracing in workflows, since I'm not entirely in the context of tracing, can you highlight them?

Reply:
Well, I have a simplistic view of it.
Pros (not specific to workflows but it applies): 
1. You can track the lifespan of a workflow, all its steps in one place. As opposed to logs where the user will have to do the join. If possible.
2. (More far fetched) You could find patterns to isolate problems in the workflow by combining metrics and traces with [examplars](https://grafana.com/docs/grafana/latest/fundamentals/exemplars/).


Con: (specific to workflows): The whole trace could become humongous and difficult to read.